File: pulse-go/internal/pulse/piece_test.go
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package pulse

import (
	"testing"
)

func TestPiece(t *testing.T) {
	t.Run("Pieces array should be valid", func(t *testing.T) {
		for piece := range Pieces {
			if Pieces[piece] != piece {
				t.Errorf("wanted pieces at index %d to be %v, got %v", piece, piece, Pieces[piece])
			}
		}
	})
}

File: pulse-kotlin/src/commonTest/kotlin/kotlin/com/fluxchess/pulse/kotlin/PieceTest.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package kotlin.com.fluxchess.pulse.kotlin

import com.fluxchess.pulse.kotlin.pieces
import kotlin.test.Test
import kotlin.test.assertEquals

class PieceTest {
	@Test
	fun `Pieces array should be valid`() {
		for (piece in pieces.indices) {
			assertEquals(pieces[piece], piece)
		}
	}
}
